packages/create-emotion-styled/types/react.d.ts

```diff
@@ -24,7 +24,7 @@ export interface StyledComponentMethods<Props extends object, InnerProps extends
   ): StyledStatelessComponent<Props, IP, Theme>;
 
   withComponent<IP extends object>(
-    component: ComponentClass<IP>,
+    component: ComponentClass<IP> | ComponentType<IP>,
```

Comment:
@mitchellhamilton It is also same in Preact, but I thought this is more clear, since this `ComponentType<IP>` is only for the library code who returns `ComponentType<IP>` in their code.